File: frontend/src/dtos/models_dtos/request_dto/JoinRequestDTO.ts


export class JoinRequestDTO {


    senderId: number
    projectPostPositionId: number
    projectPostId: number
    senderUsername: string
    id?: number
    requestedAt?: Date

    constructor(senderId: number, projectPostPositionId: number, projectPostId: number, senderUsername: string , id?: number, requestedAt?: Date) {

        this.senderId = senderId;
        this.projectPostPositionId = projectPostPositionId;
        this.projectPostId = projectPostId;
        this.senderUsername = senderUsername;
        this.id = id;
        this.requestedAt = requestedAt;
    }
}

File: backend/src/main/java/com/grouply/backend/features/join_request/dto/JoinRequestDTO.java
package com.grouply.backend.features.join_request.dto;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class JoinRequestDTO {

    Long id;
    Long senderId;
    String senderUsername;
    Long projectPostPositionId;
    Long projectPostId;
    LocalDateTime requestedAt;

}
